addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScanRuleUnitTest.java

```diff
@@ -191,6 +191,17 @@ void shouldNotAlertOnReasonableCsp() {
         assertThat(alertsRaised.size(), equalTo(0));
     }
 
+    @Test
+    void shouldNotAlertOnReasonableCspWithTrustedTypes() {
+        // Given
+        HttpMessage msg =
+                createHttpMessage("", REASONABLE_POLICY + "; require-trusted-types-for 'script'");
```

Comment:
Worth checking the other directive, also, when alerts are raised.

Reply:
Sure I can parameterize it.

Reply:
A test when alert is raised was not added though.

Reply:
Oh sorry I misunderstood. You mean one that includes trusted-types but does have other issues?

Reply:
Yes.

Reply:
Added.